applications/EigenSolversApplication/custom_solvers/eigen_dense_eigenvalue_solver.h

```diff
@@ -0,0 +1,126 @@
+/*
+//  KRATOS _______
+//        / ____(_)___ ____  ____
+//       / __/ / / __ `/ _ \/ __ \
+//      / /___/ / /_/ /  __/ / / /
+//     /_____/_/\__, /\___/_/ /_/ SolversApplication
+//             /____/
+//
+//  Author: Manuel Messmer
+*/
+
+#if !defined(KRATOS_EIGEN_DENSE_EIGENVALUE_SOLVER_H_INCLUDED)
+#define KRATOS_EIGEN_DENSE_EIGENVALUE_SOLVER_H_INCLUDED
+
+// External includes
+#include <Eigen/Core>
+#include <Eigen/Eigenvalues>
+// Project includes
+#include "includes/define.h"
+#include "spaces/ublas_space.h"
+#include "utilities/builtin_timer.h"
+#include "linear_solvers/linear_solver.h"
+
+namespace Kratos {
+
+template <typename TScalar = double,
+          class TSparseSpaceType = TUblasDenseSpace<TScalar>,
+          class TDenseSpaceType = TUblasDenseSpace<TScalar>>
+class DenseEigenvalueSolver :
```

Comment:
Maybe `DenseEigensystemSolver` to be consistent with the sparse counterpart?

Reply:
Yes your are right. I changed the naming, because this particular solver only solves for the eigenvalues of one matrix and not the eigensystem of two matrices. Would you still call it DenseEigensystemSolver?